### DISCUSSION TITLE
Poor Formatting of BigQuery Structures and Arrays

### DISCUSSION THREAD

**Discussion by sknoepfler** [dbeaver/dbeaver#26612] [Q&A]
### Description

The feature for "BigQuery: Structs and arrays are supported" was added with DBeaver 23.3.3: https://dbeaver.io/2024/01/21/dbeaver-23-3-3/

Documentation: https://dbeaver.com/docs/dbeaver/Data-View-and-Format/#structuring-complex-data-types

It does not work for me and I am unsure if its a me problem or a Dbeaver problem.

I want to visualize the following BigQuery table in dbeaver:

<img width="689" alt="image" src="https://github.com/dbeaver/dbeaver/assets/48929584/f6568242-6521-4f10-b70e-968256874603">
<img width="906" alt="image" src="https://github.com/dbeaver/dbeaver/assets/48929584/ed81c9e6-92c3-41fb-8a63-ea2bdab620cd">

In Dbeaver it looks like that even though I activated the "show complex columns" feature: 

<img width="1090" alt="image" src="https://github.com/dbeaver/dbeaver/assets/48929584/dc20a207-35e9-4024-a71e-aba6cc7f2589">

Is there anything else I have to configure in Dbeaver so that the representation of BigQuery structures and arrays work properly for me?

Thanks a lot and kind regards

### DBeaver Version

DBeaver community version: Version 23.3.4.202402060628

### Operating System

Microsoft Windows 11 Pro - 10.0.22631 Build 22631

### Database and driver

Database name and version: Google Big Query
Driver name: SimbaJDBCDriverforGoogleBigQuery42_1.5.0.1001

### Steps to reproduce

1: create the bigquery table with structures and arrays
```
CREATE OR REPLACE TABLE <project>.<dataset>.struct_and_array_table (
  id INT64,
  name STRING,
  address STRUCT<street STRING, city STRING, state STRING>,
  phone_numbers ARRAY<STRING>
);

INSERT INTO <project>.<dataset>.struct_and_array_table (id, name, address, phone_numbers)
VALUES
(1, 'John', STRUCT('123 Main St', 'Anytown', 'CA'), ['123-456-7890', '987-654-3210']),
(2, 'Alice', STRUCT('456 Elm St', 'Othertown', 'NY'), ['111-222-3333']);
```
2: connect Dbeaver with this table
3: activate the "show complex columns" feature according to the documentation: https://dbeaver.com/docs/dbeaver/Data-View-and-Format/#structuring-complex-data-types

### Additional context

- Additional extensions: none 
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

**Comment by LonwoLonwo** [comment]
Hello @sknoepfler 

I did some research about this formatting. It looks like a new feature is available only in PRO versions.

You can try it for free: https://dbeaver.com/trial/

By the way, our PRO versions provide better support for authentication methods, metadata reading, and query tracing for the BigQuery database.

**Comment by Jnb2387** [comment]
I saw that support was added in 23.3.3 in the community version, but I am still seeing the issue that sknoepfler is observing.  and in some other cases it just shows [Object]
